optimum/habana/transformers/models/llama/modeling_llama.py

```diff
@@ -372,8 +373,14 @@ def forward(
             return_dict=return_dict,
             token_idx=token_idx,
         )
-
         hidden_states = outputs[0]
+        _, seq_len, _ = hidden_states.shape
```

Comment:
This logic we can move further up just before the final norm layer.
"hidden_states = self.norm(hidden_states)"

Reply:
That could lead to some change of the output of  `output_hidden_states` and `return_dict`, while we don't want to alter the original result while this option is turned on.

Reply:
I understand. Lets do for lm head only for now where you did originally.  We can maybe enhance later.